docs/routing.md

```diff
@@ -0,0 +1,254 @@
+
+Requests that should be interpreted as requesting the `RSC` version of a page need to enter the `none` phase to receive a rewrite to the `RSC` path. However, if we assume that `check` should take us back to `filesytem`, instead of the routes that aren't in a phase, we would end up not supporting `RSC` pages properly.
+
+Therefore, when the rewritten path has changed and we are not in the `miss` phase, we send the request back to the `none` phase, so that it can be processed from the start.
+
+In the event that the rewritten path is the same as the current path, we opt to send the request to the next phase when the current phase is not `miss`. This custom check for the `miss` phase is necessary in order to prevent infite loops for requests to `/_next/static/...`, and `/_next/data/...`, as they may be rewritten to themselves when calling `check`. As such, if we are in the `miss` phase, we instead update the status code to 404.
+
+If the rewritten path was not equal to the current path but the current phase is `miss`, we instead send the request to the `filesystem` phase if the path does not exist in the build output. If it does exist, we just reset the status code if it was previously 404. This is to deal with the case where a rewrite in the `none` phase contrasts with one in the `miss` phase, e.g. with internationalization (i18n).
+
+#### Continuing with the Routing Process
+
+After all of the above has been completed, the final part to check for a (matching) route is whether it would like us to continue matching (it does so by setting its `continue` field to `true`). If it does not, we have reached the end of matching for this phase.
+
+#### Diagram
+
+```mermaid
+flowchart TD
+    checkRoute --> CheckSourceMatch{check source match}
+    CheckSourceMatch -->|true| ApplyOverrides[apply overrides]
+    CheckSourceMatch -->|false| EndRoutingSkip[`skip`]
+    ApplyOverrides --> ApplyInternationalization[apply internationalization\nredirects]
+    ApplyInternationalization --> HasMiddleware{has middleware}
+    HasMiddleware --> |yes| RunMiddlewareResult{run middleware}
+    HasMiddleware --> |no| ApplyHeadersStatusAndDestination[apply route headers\napply status code\napply destination]
+    RunMiddlewareResult --> |success| ApplyMiddlewareHeaders[apply middleware\nheaders]
+    RunMiddlewareResult --> |error| EndRoutingError[`error`]
+    ApplyMiddlewareHeaders --> ApplyHeadersStatusAndDestination
+    ApplyHeadersStatusAndDestination --> ShouldCheck{should check filesystem}
+    ShouldCheck --> |yes| PathsEqual{new path is equal}
+    PathsEqual --> |yes\n+\nphase = miss| Return404[set status to 404] --> ShouldContinue
+    PathsEqual --> |yes\n+\nphase != miss| checkNextPhase[check next phase]
+    PathsEqual --> |false| checkNonePhase[check `none` phase]
+    ShouldCheck --> |false| ShouldContinue{should continue\n and has no\nredirect}
+    ShouldContinue --> |true| ReturnNext[`next`]
+    ShouldContinue --> |false| ReturnDone[`done`]
```

Comment:
```suggestion
```mermaid
flowchart TD
    checkRoute --> CheckSourceMatch{path matches source?}
    CheckSourceMatch -->|yes| ApplyOverrides[apply overrides]
    CheckSourceMatch -->|no| EndRoutingSkip[`skip`]
    ApplyOverrides --> ApplyInternationalization[apply internationalization\nredirects]
    ApplyInternationalization --> HasMiddleware{has middleware}
    HasMiddleware --> |yes| RunMiddlewareResult{run middleware}
    HasMiddleware --> |no| ApplyHeadersStatusAndDestination[apply route headers\napply status code\napply destination]
    RunMiddlewareResult --> |success| ApplyMiddlewareHeaders[apply middleware\nheaders]
    RunMiddlewareResult --> |error| EndRoutingError[`error`]
    ApplyMiddlewareHeaders --> ApplyHeadersStatusAndDestination
    ApplyHeadersStatusAndDestination --> ShouldCheck{route.check === true}
    ShouldCheck --> |yes| PathsEqual{current path\n==\nrewritten path}
    PathsEqual --> |yes\n+\nphase = miss| Return404[set status to 404] --> ShouldContinue
    PathsEqual --> |yes\n+\nphase != miss| checkNextPhase[check next phase]
    PathsEqual --> |no| checkNonePhase[check `none` phase]
    ShouldCheck --> |no| ShouldContinue{should continue\n and has no\nredirect}
    ShouldContinue --> |yes| ReturnNext[`next`]
    ShouldContinue --> |no| ReturnDone[`done`]
```
ok then, just some minor changes 🙂